### PR TITLE
openshift_version: enterprise accepts new style pre-release

### PR DESCRIPTION
**Pull request by ashcrow**
Example: v3.7.0-0.104.0

Resolves 5192
Fixes: https://github.com/openshift/openshift-ansible/issues/5192

### CODE REVIEW

**Comment by ashcrow** [comment]
aos-ci-test

File: roles/openshift_version/tasks/main.yml [hunk 1]
```diff
@@ -60,13 +60,16 @@
...
   # - must start with a dash
   # - may contain numbers
+  # - may containe dots (https://github.com/openshift/openshift-ansible/issues/5192)
+  #
   - name: (Enterprise) Verify openshift_image_tag is valid
     when: openshift.common.deployment_type == 'openshift-enterprise'
     assert:
       that:
-      - "{{ openshift_image_tag|match('(^v\\d+\\.\\d+[\\.\\d+]*(-\\d+)?$)') }}"
+      - "{{ openshift_image_tag|match('(^v\\d+\\.\\d+[\\.\\d+]*(-[\\d+\\.]*)?$)') }}"
```
Comment (sosiouxme) [inline roles/openshift_version/tasks/main.yml]: technically this also matches `v1..3` and `v1.2.3...` and such - basically you can add a `.` anywhere.

If we care, regex might be better like:

`(^v\\d+\\.\\d+(\\.\\d+)*(-\\d+(\\.\\d+)*)?$)`

Comment (ashcrow) [inline roles/openshift_version/tasks/main.yml]: Good catch @sosiouxme

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
[merge]

**Pull request merged**